pkg/commands/plan/plan_run.go

```diff
@@ -276,55 +276,27 @@ func (c *PlanRunCommand) updateResultCommentForActions(ctx context.Context, star
 	}
 
 	c.Outf("Updating result comment")
+	msgBody := fmt.Sprintf("%s 🟦 No changes for dir: `%s` %s", CommentPrefix, c.childPath, c.gitHubLogURL)
 
-	if resulErr != nil {
+	if result.hasChanges || resultErr != nil {
 		var comment strings.Builder
-
-		fmt.Fprintf(&comment, "%s 🟥 Failed for dir: `%s` %s\n\n<details>\n<summary>Error</summary>\n\n```\n\n%s\n```\n</details>", CommentPrefix, c.childPath, c.gitHubLogURL, resulErr)
-		if result.commentDetails != "" {
-			fmt.Fprintf(&comment, "\n\n<details>\n<summary>Details</summary>\n\n```diff\n\n%s\n```\n</details>", result.commentDetails)
-		}
-
-		if err := c.gitHubClient.UpdateIssueComment(
-			ctx,
-			c.GitHubFlags.FlagGitHubOwner,
-			c.GitHubFlags.FlagGitHubRepo,
-			startComment.ID,
-			comment.String(),
-		); err != nil {
-			return fmt.Errorf("failed to update plan error comment: %w", err)
+		if result.hasChanges {
```

Comment:
Not sure if this scenarios exits, but just in case, should we invert this? In the case where hasChanges = true and err != nil, we would show success right?

Reply:
Good point. I swapped the order so that the `resultErr != nil` case takes precedence.